### ISSUE TITLE
Add basic accounting capabilities

### ISSUE THREAD

**Issue by erkarl** [ExchangeUnion/market-maker-tools#85]
This issue is about adding basic accounting capabilities to arby for the end user.

For the MVP the end user should be able to query PNL for granularity:
- all time
- past month
- this month
- last 7 days
- last 24h
- last 1h

**Comment by kilrau** [comment]
We just discussed adding a basic `/trades` endpoint which takes a date range as filter and accepts above proposed granularity, e.g. as `/trades?startDate=xyz&endDate=123&granularity=week/day/hour`.

Data fields this should return:
- opendexOrderId (xud uuid)
- cexOrderId (order id, NOT tradeIDs)
- side (buy/sell)
- amount (0.1 ETH)
- price (0.002 BTC)
- time (unix timestamp)

**Comment by kilrau** [comment]
This feature is needed for the "Closed Trades" view in the Arby/MM Bot secion in xud-ui, which only shows closed arby trades with DEX+CEX side:
![image](https://user-images.githubusercontent.com/19181985/101899730-9845ed80-3bae-11eb-85e5-4a5c402c47a1.png)

It also composes some of this info into a graph:
![image](https://user-images.githubusercontent.com/19181985/101899793-b1e73500-3bae-11eb-8164-5c07ee93d3df.png)

**Comment by erkarl** [comment]
What does **order side** mean under closed trades? It's always both - either buy on CEX and sell on DEX or vice versa.

**Comment by kilrau** [comment]
Very good point! How about now?

**Comment by erkarl** [comment]
For the chart we'll need to data in the following format:

`/profit?startDate=xyz&endDate=zyx&granularity=day`
```
{
  result: [
    { label: '1st of Jan', profit: 0.1},
    { label: '2nd of Jan', profit: 0.025},
    ...
  ]
}
```

Thinking it might make sense to have a separate endpoint for the closed trades info since it's on a separate screen and does not need granularity.

**Comment by erkarl** [comment]
> Very good point! How about now?

That works. 

Same question for the amount. Is it always the non-BTC asset?

**Comment by kilrau** [comment]
> Thinking it might make sense to have a separate endpoint for the closed trades info since it's on a separate screen and does not need granularity.

It might in future...

**Comment by erkarl** [comment]
> > Thinking it might make sense to have a separate endpoint for the closed trades info since it's on a separate screen and does not need granularity.
> 
> It might in future...

It just needs the date range though? `startDate` and `endDate`?

**Comment by kilrau** [comment]
> Same question for the amount. Is it always the non-BTC asset?

Oh damn, that took a while. So we discussed at some point to expose CEX data only here on that mm bot page along with the profit info, since all OpenDEX trade data can be looked up in `Tradehistory` and we don't want to have that double.

Updated.

**Comment by kilrau** [comment]
> > > Thinking it might make sense to have a separate endpoint for the closed trades info since it's on a separate screen and does not need granularity.
> > 
> > 
> > It might in future...
> 
> It just needs the date range though? `startDate` and `endDate`?
> 
> Or `limit` (show last 100 trades).

True. Adjusted to date range only in https://github.com/ExchangeUnion/market-maker-bot/issues/85#issuecomment-743145652

**Comment by kilrau** [comment]
I just onboarded @rsercano to this @erkarl - any questions/implementation proposal - let's track it here

**Comment by rsercano** [comment]
Hi, this's a complicated issue imo, and I guess either we need to split it into PRs, or sub issues somehow. So I was thinking to split into below MRs.

- Introducing sequelize and a new db type `arby-order.ts`
- Saving CEX orders into db after converting to `arby-order.ts` async. (Open to suggestions about async process, if it can be done on rx-js)
- Introducing http webserver to arby.
- Implementing new `/trades` endpoint as explained above.

wdyt? @erkarl @kilrau

**Comment by erkarl** [comment]
> Introducing sequelize and a new db type arby-order.ts

Let's start with adding sequelize support and define what we're going to save to the database.

**Comment by rsercano** [comment]
Edited my comment to be brief, @erkarl started to implement db model

**Comment by kilrau** [comment]
As just discussed with @erkarl and @rsercano : 
- we can't link each Binance trade with an OpenDEX trade directly (e.g. arby might accumulate xud trades if amount is too small to execute on Binance and later issue one binance order covering multiple xud trades)
- instead we'll use the ccxt order output and save it to a binance trades db as is
- profit will be only be calculatable for a given time frame: sum up (price x amount) binance & xud, subtract binance from xud = profit

Removed Wireframe adjusted too.
